extension/src/popup/views/SignTransaction/hooks/useGetSignTxData.tsx

```diff
@@ -130,10 +131,14 @@ function useGetSignTxData(
       dispatch({ type: "FETCH_DATA_SUCCESS", payload: firstRenderPayload });
 
       // Add all icons needed for tx assets
-      const icons = {} as { [code: string]: string | null };
-      let assetsListsData: AssetListResponse[] = [];
+      const icons = {} as { [code: string]: string };
+      // Initialize with cached lists, but we'll fetch fresh data when needed for icon lookups
+      let assetsListsData: AssetListResponse[] = cachedTokenLists || [];
+      let hasFetchedAssetLists = false;
```

Comment:
should this be `hasFetchedAssetLists = assetsListsData.length > 0`? Looks like you could already have the assetLists if `cachedTokenLists` is populated